sklearn/mixture/tests/test_bayesian_mixture.py

```diff
@@ -419,3 +423,46 @@ def test_invariant_translation():
             assert_almost_equal(bgmm1.means_, bgmm2.means_ - 100)
             assert_almost_equal(bgmm1.weights_, bgmm2.weights_)
             assert_almost_equal(bgmm1.covariances_, bgmm2.covariances_)
+
+
+def test_bayesian_mixture_fit_predict():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng, scale=7)
+    n_components = 2 * rand_data.n_components
+
+    for covar_type in COVARIANCE_TYPE:
+        bgmm1 = BayesianGaussianMixture(n_components=n_components,
+                                        max_iter=100, random_state=rng, tol=1e-3,
+                                        reg_covar=0)
+        bgmm1.covariance_type = covar_type
+        bgmm2 = copy.deepcopy(bgmm1)
+        X = rand_data.X[covar_type]
+
+        Y_pred1 = bgmm1.fit(X).predict(X)
+        Y_pred2 = bgmm2.fit_predict(X)
+        assert_array_equal(Y_pred1, Y_pred2)
+
+
+def test_bayesian_mixture_predict_predict_proba():
```

Comment:
If this was copied from the other test, can you maybe say so?

Reply:
I don't get how this relates to the issue... What am I missing?

Reply:
Sorry just saw this! This was copied from other test. I will comment on that. We added this test because to test fit_predict, we intended to test two things: 1. it is equivalent to fit().predict(); 2. it's output is correct. There was no testing for correctness of predict() for bgmm, we added one so that we know fit_predict actually yields the correct output.

Reply:
I see, great!